Router.svelte

```diff
@@ -49,6 +49,53 @@ function getLocation() {
     return {location, querystring}
 }
 
+/**
+ * Transforms the path into a RegExp pattern and
+ *  extracts the named params (keys)
+ * 
+ * @param {string | RegExp} path - Path from route config
+ * @property {RegExp} pattern - Resulting regex pattern
+ * @property {string[]} keys - Named params of the path
+ * 
+ * @returns {{pattern: RegExp, keys: string[]}}
+ * @private
+ */
+function parsePath(path) {
+    if (typeof path == 'string') {
+        return regexparam(path)
+    }
+    // Else path is a RegExp
+
+    const keys = []
+
+    // Named parameters in a RegExp path are represented with capture groups (`?<parameter_name>`).
+    // We need to extract the inside as key, and add an almost-wildcard
+    //  expression `([^\\/]+?)` that will match like a string path param.
+    let newPath = path.toString()
+        .replace(/[?]<.*>/g, (match) => {
```

Comment:
I would really prefer if this were an actual parser than some regular expression magic. Parsers built using a regexp are generally frowned upon because they might not be the most robust.

How about parsing with just a for loop iterating on every character, looking for the `(?` sequence?

Reply:
Okok, I will do some search on how such a parser would ideally be implemented, I guess it should not be that hard ;p

Reply:
Would a BNF state-machine be something interesting in that case? I'm thinking about something like this :

<details>
<summary>Algorithm</summary>

```js
function parseRegexPath(path) {
    const keys = []
    const newPathCharacters = []
    const KEY_START_1 = '?'
    const KEY_START_2 = '<'
    const KEY_END = '>'
    const PATH_PARAM_REGEX = '([^\\/]+?)'

    let keyBuffer = []
    let state = STATE_NOT_KEY // some Enum, TS Union or set of constants

    path.toString().split('').forEach(character => {
        newPathCharacters.push(character)

        if (state === STATE_NOT_KEY && character === KEY_START_1) {
            state = STATE_MAYBE_KEY
        }
        else if (state === STATE_MAYBE_KEY && character === KEY_START_2) {
            state = STATE_KEY
        }
        else if (state === STATE_KEY) {
            if (character === KEY_END) {
                keys.push(keyBuffer.join(''))
                newPathCharacters.push(PATH_PARAM_REGEX)
                state = STATE_NOT_KEY
                keyBuffer = []
            } else {
                keyBuffer.push(character)
            }
        }
    })

    let newPath = newPathCharacters.join('')

    // ... some other final logic and return
}
```

</details>